site/content/entry/citra-progress-report-2017-p1.md

```diff
@@ -0,0 +1,33 @@
+
+For now, you can only configure your gamepad by manually editing a config file, but [JayFoxRox](https://github.com/jayfoxrox) has written a little tool to help edit the config file while we integrate it better into Citra's UI. You can find the tool in the Citra forums [here](https://community.citra-emu.org/t/temporary-controller-configurations-for-citra/1061).
+
+## [More Shader Refactoring Part 1](https://github.com/citra-emu/citra/pull/2335), [Part 2](https://github.com/citra-emu/citra/pull/2346), & [Part 3](https://github.com/citra-emu/citra/pull/2476) by [yuriks](https://github.com/yuriks)
+
+Citra's video code has grown organically from the start, not much was known about the PICA200 (i.e., the 3DS GPU), so a lot of the backend for it was based on pure guesswork and the team's past experience with working with emulation. It made a lot of assumptions on how it worked internally, which worked okay some of the time, but it was amazing at the time to even have a 3DS game load, let alone render a scene. So it was left alone.
+
+The entire architecture of the 3DS was largely unknown, so development was done in small seperate bits at a time, and then trying to piece it all together into one cohesive piece of software. But, over time, as new pieces of information were found, the assumptions that were made previously broke. Some assumptions were very large, obvious, and easy to overcome, but a more subtle error that these assumptions created were not even in the design of the GPU, but rather the architecture and organization of the code itself.
+
+The video code was very interconnected using this interface, so despite people being aware of this, it was also extremely difficult to move code around and reorganize it, as it could very easily affect unrelated bits of code that depended on it. This was only compounded by the hardware renderer, as it was written much like earlier, slowly adding hardware support bit-by-bit to seperate parts of the video code, and simply falling back into the software renderer when disabled or not supported.
+
+There aren't many changes visible on the surface here, but internally this rewrite of the video code that [yuriks](https://github.com/yuriks) has been working on for the past few months has laid the foundation that Citra needs to be able to grow and mature in a clean and maintainable way. With these, things like geometry shaders (which many games heavily depend upon), more support for hardware renderers, and more accurate software renderers, could all be added with little to no effort besides the actual implementation.
+
+## [UDS Services](https://github.com/citra-emu/citra/pull/2628) by [Subv](https://github.com/Subv)
```

Comment:
Yeah, @JayFoxRox mentioned both @Subv potentially not wanting hype just yet, and I also told him that and said that the LUT images are there because I'd also written an article in 70fb0b9, but I found it to be a bit awkwardly worded, and too short, since it was a mainly internal thing.